refactoring/intersection.js

```diff
@@ -0,0 +1,16 @@
+// Find an intersection of two dictionaries
+
+'use strict';
+
+const intersection = (object1, object2) => {
+  const intersecObj = {};
+  const firstKeys = Object.keys(object1);
+  for (const key of firstKeys) {
+    if (object1[key] === object2[key]) {
+      intersecObj[key] = object1[key];
```

Comment:
Duplicate expression `object1[key]` it's better to put it into intermediate variable